### PR TITLE
fix revision history broken bug

### PR DESCRIPTION
**Pull request by rahul2240**
Fixes https://github.com/openSUSE/osem/issues/2334

### CODE REVIEW

File: app/controllers/admin/versions_controller.rb [hunk 1]
```diff
@@ -6,12 +6,15 @@ class VersionsController < Admin::BaseController
     load_and_authorize_resource class: PaperTrail::Version
 
     def index
-      @conferences_with_role = current_user.is_admin? ? Conference.pluck(:short_title) : Conference.with_role([:organizer, :cfp, :info_desk], current_user).pluck(:short_title)
-
+      if Conference.any?
```
Comment (Ana06) [inline app/controllers/admin/versions_controller.rb]: why is this needed? how is this fixing the bug?

File: app/controllers/admin/versions_controller.rb [hunk 2]
```diff
@@ -6,12 +6,15 @@ class VersionsController < Admin::BaseController
...
+        @conferences_with_role = current_user.is_admin? ? Conference.pluck(:short_title) : Conference.with_role([:organizer, :cfp, :info_desk], current_user).pluck(:short_title)
+        @conferences_with_role.uniq!
+      end
       if current_user.has_cached_role? :organization_admin, :any
         @conferences_with_role = Organization.with_role('organization_admin', current_user).map { |org| org.conferences.pluck :short_title }.flatten
+        @conferences_with_role.uniq!
       end
-      @conferences_with_role.uniq!
```
Comment (Ana06) [inline app/controllers/admin/versions_controller.rb]: why did you need to move this line?

File: app/views/admin/versions/index.html.haml [hunk 3]
```diff
@@ -8,8 +8,9 @@
           %span.caret
         %ul.dropdown-menu
           %li= link_to 'All Conferences & Users',  admin_revision_history_path
-          - @conferences_with_role.each do |conference_short_title|
-            %li= link_to conference_short_title,  admin_conference_revision_history_path(conference_id: conference_short_title)
+          - if !@conferences_with_role.nil?
```
Comment (Ana06) [inline app/views/admin/versions/index.html.haml]: this can only be `nil` because `@conferences_with_role.uniq!` was moved.

**Comment by rahul2240** [comment]
@Ana06 I misunderstood the issue, my bad. When I was working on it, I got undefined method for nil:nilclass error in revision history so i thought, it is due to Conference is nil and checking for the conditions as shown in this commit I fixed that but I am no longer able to recreate that, also revision history link in dropdown is working for me.
